src/main/java/hudson/plugins/git/GitPublisher.java

```diff
@@ -500,18 +514,34 @@ public void setEmptyTargetRepoToOrigin(){
     }
 
     public static final class BranchToPush extends PushConfig {
-        private String branchName;
 
-        public String getBranchName() {
-            return branchName;
-        }
+        private String srcBranchName;
+        private String branchName;   // dstBranchName
 
         @DataBoundConstructor
-        public BranchToPush(String targetRepoName, String branchName) {
+        public BranchToPush(String targetRepoName, String srcBranchName, String branchName) {
```

Comment:
You may get :-1: from my previous employer.
I definitely used it somewhere in our validated merge flows